python/src/equistore/block.py

```diff
@@ -205,6 +205,52 @@ def gradient(self, parameter: str) -> "Gradient":
         Get the gradient of the ``values`` in this block with respect to
         the given ``parameter``.
 
+        Here is an example of how to use this method:
+        >>> import numpy as np
+        >>> block_1 = TensorBlock(
+        ...             values=np.full((3, 1, 1), 1.0),
+        ...             samples=Labels(
+        ...                 ["samples"], np.array([[0], [2], [4]], dtype=np.int32)
+        ...             ),
+        ...             components=[Labels(
+        ...                 ["components"], np.array([[0]], dtype=np.int32))
+        ...             ],
+        ...             properties=Labels(
+        ...                 ["properties"], np.array([[0]], dtype=np.int32)),
+        ...         )
+        >>> block_1.add_gradient(
+        ...     "parameter1",
+        ...     samples=Labels(
+        ...        ["sample", "parameter1"], np.array([[0, -2], [2, 3]], dtype=np.int32)
+        ...     ),
+        ...     data=np.full((2, 1, 1), 11.0),
+        ...     components=[Labels(["components"], np.array([[0]], dtype=np.int32))],
+        ... )
+        >>> block_1.add_gradient(
+        ...     "parameter2",
+        ...     samples=Labels(
+        ...        ["sample", "parameter2"], np.array([[0, -2], [2, 3]], dtype=np.int32)
+        ...     ),
+        ...     data=np.full((2, 1, 1), 15.0),
+        ...     components=[Labels(["components"], np.array([[0]], dtype=np.int32))],
+        ... )
```

Comment:
It would be helpful if we can show in the second block how to add components in gradients,
```suggestion
        >>> block_1.add_gradient(
        ...     "cell",
        ...     data=np.full((2, 6, 1, 1), 15.0),
        ...     samples=Labels(
        ...        ["sample"], np.array([[0], [2]], dtype=np.int32)
        ...     ),
        ...     components=[Labels(["cell"], np.array([[0], [1], [2], [3], [4], [5]], dtype=np.int32)), Labels(["components"], np.array([[0]], dtype=np.int32))],
        ... )
```